core/src/main/java/greencity/service/impl/EcoNewsServiceImpl.java

```diff
@@ -76,6 +79,13 @@ public AddEcoNewsDtoResponse save(AddEcoNewsDtoRequest addEcoNewsDtoRequest,
         if (image != null) {
             toSave.setImagePath(fileService.upload(image).toString());
         }
+
+        Set<String> tagsSet = new HashSet<>(addEcoNewsDtoRequest.getTags());
+
```

Comment:
Could you please explain this if condition?

- `tagsSet `is created as a copy of `addEcoNewsDtoRequest.getTags()`
- so the size of `tagsSet `is the same as the size of `addEcoNewsDtoRequest.getTags()`
- so the if condition is always false
- so we have an unreachable statement?

Reply:
addEcoNewsDtoRequest.getTags() this is List which can contain duplicates of tags that not allowed. So I created a Set which will have smaller size than  addEcoNewsDtoRequest.getTags() if it contains duplicates

Reply:
Does it makes sense to save the Set without duplicates and do no return the error to user?  To make the correction for user on our own?

Reply:
In if condition I throw exception, so user will be informed about error. Also I thought about auto correction, but I made it as "Expected result" was required.

Reply:
Could you please give me a link to find the "Expected result"?

Reply:
https://github.com/ita-social-projects/GreenCity/issues/718